### ISSUE TITLE
Error: The non-abstract class '_ZefyrSelectionOverlayState' is missing implementations for these members:

### ISSUE THREAD

**Issue by henaff** [memspace/zefyr#121]
## Steps to Reproduce

Just install zefyr package and setup the minimal required to use the editor.

```yaml
zefyr:
    git:
      url: https://github.com/memspace/zefyr.git
      ref: flutter-master
      path: packages/zefyr
```

```dart
import 'package:flutter/material.dart';
import 'package:zefyr/zefyr.dart';

class PostScreen extends StatefulWidget {
  @override
  _PostScreenState createState() => _PostScreenState();
}

class _PostScreenState extends State<PostScreen> {

  ZefyrController _controller;
  FocusNode _focusNode;

  @override
  void initState() {
    super.initState();
    // Create an empty document or load existing if you have one.
    // Here we create an empty document:
    final document = new NotusDocument();
    _controller = new ZefyrController(document);
    _focusNode = new FocusNode();
  }

  @override
  Widget build(BuildContext context) {
    return ZefyrScaffold(
      child: ZefyrEditor(
        controller: _controller,
        focusNode: _focusNode,
      ),
    );
  }

}
```

## Logs

```none
Compiler message:
file:///home/henaff/.pub-cache/git/zefyr-3be80cf9c24abb0388daaf0baaad28a2c09f5220/packages/zefyr/lib/src/widgets/selection.dart:42:7: Error: The non-abstract class '_ZefyrSelectionOverlayState' is missing implementations for these members:
 - TextSelectionDelegate.copyEnabled
 - TextSelectionDelegate.cutEnabled
 - TextSelectionDelegate.pasteEnabled
 - TextSelectionDelegate.selectAllEnabled
Try to either
 - provide an implementation,
 - inherit an implementation from a superclass or mixin,
 - mark the class as abstract, or
 - provide a 'noSuchMethod' implementation.

class _ZefyrSelectionOverlayState extends State<ZefyrSelectionOverlay>
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^
file:///opt/flutter/packages/flutter/lib/src/services/text_input.dart:596:12: Context: 'TextSelectionDelegate.copyEnabled' is defined here.
  bool get copyEnabled => true;
           ^^^^^^^^^^^
file:///opt/flutter/packages/flutter/lib/src/services/text_input.dart:593:12: Context: 'TextSelectionDelegate.cutEnabled' is defined here.
  bool get cutEnabled => true;
           ^^^^^^^^^^
file:///opt/flutter/packages/flutter/lib/src/services/text_input.dart:599:12: Context: 'TextSelectionDelegate.pasteEnabled' is defined here.
  bool get pasteEnabled => true;
           ^^^^^^^^^^^^
file:///opt/flutter/packages/flutter/lib/src/services/text_input.dart:602:12: Context: 'TextSelectionDelegate.selectAllEnabled' is defined here.
  bool get selectAllEnabled => true;
           ^^^^^^^^^^^^^^^^
file:///home/henaff/.pub-cache/git/zefyr-3be80cf9c24abb0388daaf0baaad28a2c09f5220/packages/zefyr/lib/src/widgets/rich_text.dart:184:26: Error: The parameter 'value' of the method 'RenderZefyrParagraph.text' has type 'TextSpan', which does not match the corresponding type, 'InlineSpan', in the overridden method, 'RenderParagraph.text'.
 - 'TextSpan' is from 'package:flutter/src/painting/text_span.dart' ('file:///opt/flutter/packages/flutter/lib/src/painting/text_span.dart').
 - 'InlineSpan' is from 'package:flutter/src/painting/inline_span.dart' ('file:///opt/flutter/packages/flutter/lib/src/painting/inline_span.dart').
Change to a supertype of 'InlineSpan', or, for a covariant parameter, a subtype.
  void set text(TextSpan value) {
                         ^
file:///opt/flutter/packages/flutter/lib/src/rendering/paragraph.dart:115:7: Context: This is the overridden method ('text').
  set text(InlineSpan value) {
      ^
file:///home/henaff/.pub-cache/git/zefyr-3be80cf9c24abb0388daaf0baaad28a2c09f5220/packages/zefyr/lib/src/widgets/editable_text.dart:102:28: Error: The method 'reparentIfNeeded' isn't defined for the class 'FocusScopeNode'.
 - 'FocusScopeNode' is from 'package:flutter/src/widgets/focus_manager.dart' ('file:///opt/flutter/packages/flutter/lib/src/widgets/focus_manager.dart').
Try correcting the name to the name of an existing method, or defining a method named 'reparentIfNeeded'.
    FocusScope.of(context).reparentIfNeeded(focusNode);
                           ^^^^^^^^^^^^^^^^
file:///home/henaff/.pub-cache/git/zefyr-3be80cf9c24abb0388daaf0baaad28a2c09f5220/packages/zefyr/lib/src/widgets/selection.dart:483:22: Error: Too few positional arguments: 6 required, 4 given.
        .buildToolbar(context, editingRegion, midpoint, widget.delegate);
                     ^
file:///home/henaff/.pub-cache/git/zefyr-3be80cf9c24abb0388daaf0baaad28a2c09f5220/packages/zefyr/lib/src/widgets/input.dart:153:18: Error: The argument type 'String' can't be assigned to the parameter type 'DiagnosticsNode'.
 - 'DiagnosticsNode' is from 'package:flutter/src/foundation/diagnostics.dart' ('file:///opt/flutter/packages/flutter/lib/src/foundation/diagnostics.dart').
Try changing the type of the parameter, or casting the argument to 'DiagnosticsNode'.
        context: 'while updating editing value',
                 ^
Compiler failed on /home/henaff/Documents/Programming/Projects/FIXR/fixr_v1/lib/main.dart

FAILURE: Build failed with an exception.

* Where:
Script '/opt/flutter/packages/flutter_tools/gradle/flutter.gradle' line: 652

* What went wrong:
Execution failed for task ':app:compileflutterBuildDebugArm'.
> Process 'command '/opt/flutter/bin/flutter'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 15s
Gradle task assembleDebug failed with exit code 1
Exited (sigterm)
```

```
[✓] Flutter (Channel master, v1.7.11-pre.48, on Linux, locale en_US.UTF-8)
    • Flutter version 1.7.11-pre.48 at /opt/flutter
    • Framework revision e5e16406e5 (18 hours ago), 2019-06-28 09:27:55 -0400
    • Engine revision 5c3c433639
    • Dart version 2.4.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /home/henaff/Android/Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b16-5323222)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find chrome executable at google-chrome)
    ! CHROME_EXECUTABLE not set

[✓] Linux toolchain - develop for Linux desktop
    • clang++ 8.0.0
    • GNU Make 4.2.1

[✓] Android Studio (version 3.4)
    • Android Studio at /opt/android-studio
    • Flutter plugin version 36.0.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b16-5323222)

[✓] Connected device (3 available)
    • SM G530FZ • 785d3a56 • android-arm    • Android 5.0.2 (API 21)
    • Linux     • Linux    • linux-x64      • Linux
    • Chrome    • chrome   • web-javascript • unknown (unsupported)

! Doctor found issues in 1 category.
```

**Comment by gedw99** [comment]
Is the same error occuring on Flutter Dev channel ?

**Comment by fuaditrockz** [comment]
@henaff Just uninstall and install it again. It work for me so well.

My version is:
    Dart: 2.4.0
    pana: 0.12.19
    Flutter: 1.7.8+hotfix.3

**Comment by pulyaevskiy** [comment]
This should be fixed in just released 0.6.0.
